### PR TITLE
Merging forward and backward kernels into one

### PR DESCRIPTION
**Pull request by novakovicdj**
Replaced two kernels with one because they are similar. 
Merged two dropout functions that are calling kernels into one with parameter is_backward to determine which kernel should be executed.

### CODE REVIEW

File: src/dropout_api.cpp [hunk 1]
```diff
@@ -187,14 +187,15 @@ extern "C" miopenStatus_t miopenDropoutForward(miopenHandle_t handle,
...
+            .Dropout(miopen::deref(handle),
                             miopen::deref(noise_shape),
                             miopen::deref(xDesc),
                             DataCast(x),
                             miopen::deref(yDesc),
                             DataCast(y),
                             DataCast(reserveSpace),
-                            reserveSpaceSizeInBytes);
+                            reserveSpaceSizeInBytes,
+                            false /* is_backward */);
```
Comment (junliume) [inline src/dropout_api.cpp:201]: @novakovicdj some HIP tidy issues:
```
dropout_api.cpp:198:29: error: implicit conversion 'bool' -> 'size_t' (aka 'unsigned long') [readability-implicit-bool-conversion,-warnings-as-errors]
```

File: src/dropout_api.cpp [hunk 2]
```diff
@@ -213,13 +214,14 @@ extern "C" miopenStatus_t miopenDropoutBackward(miopenHandle_t handle,
...
+            .Dropout(miopen::deref(handle),
                              miopen::deref(noise_shape),
                              miopen::deref(dyDesc),
                              DataCast(dy),
                              miopen::deref(dxDesc),
                              DataCast(dx),
                              DataCast(reserveSpace),
-                             reserveSpaceSizeInBytes);
+                             reserveSpaceSizeInBytes,
+                             true /* is_backward */);
```
Comment (junliume) [inline src/dropout_api.cpp:231]: Ditto

**Comment by junliume** [comment]
@novakovicdj could you help with the above mentioned issue? Thanks!

**Comment by novakovicdj** [comment]
@junliume hi, can you review this one again? Thanks

**Review by junliume — APPROVED** [review]

**Pull request merged**